### DISCUSSION TITLE
Creating a common representation of Model (Wing Type/Tail Type, Mix) to help interoperability between differen tools

### DISCUSSION THREAD

**Discussion by frankiearzu** [EdgeTX/edgetx#2905] [LUA]
Currently, Im working on DSM Forward Programming tool for Spektrum receivers with Gyro (Stabilization, AutoLevel, Envelope, Panic),
FySky SxR Reveivers could use similar functionality, and probably other flight controllers.

The RX needs to know where are your ailerons (primary, and slave),  elevator/rudder (one or two of each), as well as if you have any special mix for them (Delta Wing, Vtail or Ailevator tail).  In that way the RX can "fly" the plane.

Without a common model Wing/Tail type representation that can be shared across tools/scripts, you will have to setup similar things in multiple different scripts.

I think there are two framework things here:  
1. Common metadata to describe the model Wing and Tail, with the ability to setup your own channels/mixers.
2. Model Creation/Editor tools to do this configuration
2.a) Wizard to setup your servos and mixes on your own (Create as well as Modify)
2.b) Pre-Configured model templates like the JustFly or the RCSoar templates (with servos assigned, and other things).

By having that common metadata representation, the Gyro tool can use that data and automatically setup itself.

For example, imagine that Wizard/JustFly/RCSoar can create the metadata when setting up the model,  The DSM Forward Programming tool can take that metadata and just use it.    

Currently, the Gyro script (Spektrum or FrSky) will need to provide its own way to descrive the plane to the gyro!  Double work.
On radios like Spektrum/Futaba, all this is integrated, so you don't need to redo things, but they are close systems, in this case, it will be an "Open" version of it.

Examples of my Gyro Setup screens for DSM Forward Programming: 
![Capture5](https://user-images.githubusercontent.com/32604366/208480800-7ecdce20-83c9-4701-9934-ae55f5067240.JPG)
![Capture2](https://user-images.githubusercontent.com/32604366/208480813-643dd8d6-e131-4dcf-8a4a-2f71ce96ff0c.JPG)
![Capture3](https://user-images.githubusercontent.com/32604366/208480824-92db6cff-90cc-4f40-8c46-2dc77682c00e.JPG)
![Capture4](https://user-images.githubusercontent.com/32604366/208480836-685d6bb4-0a1d-4720-8686-8815e89b072c.JPG)

**Comment by frankiearzu** [comment]
@lshems   I also will be open to work on a common representation.  That can be shared between the Wizard or Template base creation tools.    Later, the Gyro tools or Flight controllers can use that data to start their setup without duplicating effort.

**Reply by lshems** [reply]
Super

**Reply by lshems** [reply]
Also, there is this glider sim from Mike. It shows a model plane in a 2D projection from behind, and shows the outputs on the model surfaces as well. It needs the same standard conventions.

https://repository.justfly.solutions/index.php/lua-scripts?view=product&id=14:glidsim
